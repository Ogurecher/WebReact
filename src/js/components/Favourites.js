import React, { Component } from "react";
import { CityInfo } from "./CityInfo";

export default class Favourites extends Component {
    
    render() {
        const favourites = localStorage.getItem('favourites').split(',').map((element) =>               //update component on localstorage update
            <div key={element.toString()}>
                <CityInfo />
            </div>
        );
        return (
            <div>
                {favourites}
            </div>
        );
    }
}