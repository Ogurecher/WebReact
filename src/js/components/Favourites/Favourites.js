import React, { Component } from 'react';
import CityInfo from '../CityInfo/CityInfo';
import './Favourites.css';

export default class Favourites extends Component {
    
    render() {
        let cities = this.props.cities.slice();
        cities = cities.reduce((unique, elem1) => {
            if(!unique.some(elem2 => elem2.position.city === elem1.position.city)) {
              unique.push(elem1);
            }
            return unique;
        },[]);
        cities = cities.map((element, index) => {
            if (this.props.favourites.includes(element.position.city.toString())) {
                return (<CityInfo key={index} city={element} />);
            }
            return null;
        }
                
        );
        return (
            <div className='favourites-cities-wrapper'>
                {cities}
            </div>
        );
    }
}