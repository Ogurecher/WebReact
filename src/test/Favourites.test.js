import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Favourites from '../js/components/Favourites/Favourites';

const store = configureMockStore();

describe('Favourites', function() {
    test('snapshot renders with empty favourites', function() {
        const citiesStub = [];
        const favouritesStub = [];

        const component = renderer.create(<Favourites cities={citiesStub} favourites={favouritesStub} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('snapshot renders with non-empty favourites', function() {
        const weatherStub = {
            name: 'London',
            wind: {
                deg: 'winddeg',
                speed: 'windspd'
            },
            weather: [{
                description: 'desc',
                icon: '50n'
            }],
            main: {
                pressure: 'pressure',
                humidity: 'humidity'
            },
            coord: {
                lat: 'lat',
                lon: 'lon'
            }
        };

        const cityStub = {
            position: {
                city: 'London'
            },
            weather: weatherStub,
            loading: 0,
            errorMsg: null
        };
        const citiesStub = [cityStub];
        const favouritesStub = ['London'];


        const storeMock = store();

        const component = renderer.create(<Provider store={storeMock}><Favourites cities={citiesStub} favourites={favouritesStub} /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});