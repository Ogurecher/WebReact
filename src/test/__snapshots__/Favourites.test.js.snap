// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites snapshot renders with empty favourites 1`] = `
<div
  className="favourites-cities-wrapper"
/>
`;

exports[`Favourites snapshot renders with non-empty favourites 1`] = `
<div
  className="favourites-cities-wrapper"
>
  <div
    className="city-info"
  >
    <div
      className="city-info-head"
    >
      <p>
         
        London
         
      </p>
      <img
        src="http://openweathermap.org/img/wn/50n@2x.png"
      />
      <p>
         
        Â°C 
      </p>
      <form
        onSubmit={[Function]}
      >
        <input
          type="submit"
          value="X"
        />
      </form>
    </div>
    <div
      className="city-info-weather-list"
    >
      <div
        className="city-info-weather-list-item"
      >
        <p>
           
          Wind
           
        </p>
        <p>
           
          winddegdeg
          ; windspd m/s
           
        </p>
      </div>
      <div
        className="city-info-weather-list-item"
      >
        <p>
           
          Clouds
           
        </p>
        <p>
           
          desc
           
        </p>
      </div>
      <div
        className="city-info-weather-list-item"
      >
        <p>
           
          Pressure
           
        </p>
        <p>
           
          pressure hpa
           
        </p>
      </div>
      <div
        className="city-info-weather-list-item"
      >
        <p>
           
          Humidity
           
        </p>
        <p>
           
          humidity%
           
        </p>
      </div>
      <div
        className="city-info-weather-list-item"
      >
        <p>
           
          Coordinates
           
        </p>
        <p>
           
          [lat
          ; lon]
           
        </p>
      </div>
    </div>
  </div>
</div>
`;
